### PR TITLE
Mono repo implementation

### PR DESCRIPTION
**Pull request by ghost**
# Description

Implements yarn workspace for orakl repo

Checked yarn commands, docker build commands, and github actions including tests

## yarn workspace commands

Package commands can now be executed from root directory by calling
`yarn {Package Name} {Package Yarn Command}`

ex. `yarn contracts clean` , `yarn cli build`, `yarn api build`, and etc..

You can check how this command is mapped through workspace command in root `package.json` file

## removal of local `yarn.lock` files for each folders

Since dependencies are hoisted into root folder's `yarn.lock` file, `yarn.lock` files in each project folders are removed.

## local package reference

`@bisonai/orakl-vrf` & `@bisonai/contracts` are now referenced through local package. If there are dependency, you should first build inside each project folder. For example, if you are trying to build `core` from your environment, you should build `contracts` first. it can be checked through github action yaml files. you can check this kind of dependency by calling `yarn workspaces info` from root dir.

## further works to be done

These tasks will be done from further PRs.

- Dockerfile updates to build and run from root dir, currently just removed copying yarn.lock file for compatibility
- Remove symlink as written in issue (https://github.com/Bisonai/orakl/issues/937#issuecomment-1822142313)

## "*" version referencing for inner workspace dependency

no explicit version written in the package.json file, but it could be specified later

### warning

- admin project has been excluded from mono repo.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

### CODE REVIEW

File: contracts/test.txt
Comment (martinkersner) [inline contracts/test.txt:1]: @nick-bisonai Let's remove this test file before merging 🙌

File: contracts/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "skipLibCheck": true,
```
Comment (martinkersner) [inline contracts/tsconfig.json:3]: @nick-bisonai Is this also related to the hardhat issue?

**Review by martinkersner — APPROVED** [review]
LGTM!

File: contracts/tsconfig.json (hunk 1 above)
Comment (nick-bisonai) [reply contracts/tsconfig.json:3]: I added this line to fix build error which happened after mono repo implementation. it was type some kind of type error

```
../node_modules/@types/mocha/index.d.ts:2642:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'beforeEach' must be of type 'Lifecycle', but here has type 'HookFunction'.

2642 declare var beforeEach: Mocha.HookFunction;
                 ~~~~~~~~~~

  ../node_modules/@types/jest/index.d.ts:34:13
    34 declare var beforeEach: jest.Lifecycle;
                   ~~~~~~~~~~
    'beforeEach' was also declared here.

../node_modules/@types/mocha/index.d.ts:2660:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'afterEach' must be of type 'Lifecycle', but here has type 'HookFunction'.

2660 declare var afterEach: Mocha.HookFunction;
                 ~~~~~~~~~

  ../node_modules/@types/jest/index.d.ts:36:13
    36 declare var afterEach: jest.Lifecycle;
                   ~~~~~~~~~
    'afterEach' was also declared here.

../node_modules/@types/mocha/index.d.ts:2676:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'describe' must be of type 'Describe', but here has type 'SuiteFunction'.

2676 declare var describe: Mocha.SuiteFunction;
                 ~~~~~~~~

  ../node_modules/@types/jest/index.d.ts:37:13
    37 declare var describe: jest.Describe;
                   ~~~~~~~~
    'describe' was also declared here.

../node_modules/@types/mocha/index.d.ts:2697:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'xdescribe' must be of type 'Describe', but here has type 'PendingSuiteFunction'.

2697 declare var xdescribe: Mocha.PendingSuiteFunction;
                 ~~~~~~~~~

  ../node_modules/@types/jest/index.d.ts:39:13
    39 declare var xdescribe: jest.Describe;
                   ~~~~~~~~~
    'xdescribe' was also declared here.

../node_modules/@types/mocha/index.d.ts:2711:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'it' must be of type 'It', but here has type 'TestFunction'.

2711 declare var it: Mocha.TestFunction;
                 ~~

  ../node_modules/@types/jest/index.d.ts:40:13
    40 declare var it: jest.It;
                   ~~
    'it' was also declared here.

../node_modules/@types/mocha/index.d.ts:2725:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'test' must be of type 'It', but here has type 'TestFunction'.

2725 declare var test: Mocha.TestFunction;
                 ~~~~

  ../node_modules/@types/jest/index.d.ts:43:13
    43 declare var test: jest.It;
                   ~~~~
    'test' was also declared here.

../node_modules/@types/mocha/index.d.ts:2732:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'xit' must be of type 'It', but here has type 'PendingTestFunction'.

2732 declare var xit: Mocha.PendingTestFunction;
                 ~~~

  ../node_modules/@types/jest/index.d.ts:42:13
    42 declare var xit: jest.It;
                   ~~~
    'xit' was also declared here.


Found 7 errors in the same file, starting at: ../node_modules/@types/mocha/index.d.ts:2642
```

Comment (martinkersner) [reply contracts/tsconfig.json:3]: Okay! Thank you!
